qcore.c

```diff
@@ -11,132 +11,89 @@ DESCRIPTION:	The core of the QUERK kernel. Includes the software interrupt
 
 #include "qcore.h"
 #include "queue.h"
-#include "qrequest.h"
+#include "quser.h"
 #include "nios2_ctrl_reg_macros.h"
 
-/*----------------------------------------------------------------*/
+static Queue ready_queue;
+static Process *running_process;
+static Process process_array[MAX_NUM_OF_PROCESSES];
 
-static  Process     *running_process;
+static char kernel_stack[512];
+static void *kernel_stack_pointer = (void *) &kernel_stack[511];
 
-static  Queue   ready_queue;
+static int num_of_processes = 0;
+static int need_dispatch;
 
-volatile int* JTAG_UART_ptr = (int*) 0x10001000;// JTAG UART address
-
-static  char    kernel_stack[512];
-static  void    *kernel_stack_pointer = (void *) &kernel_stack[511];
-// Ask why we can't just define this in the exception handler and
-
-static  int     num_of_processes = 0;	/* counter used to assign unique pids */
-
-static  Process     process_array[MAX_NUM_OF_PROCESSES]; // Formerly pdb_array
 extern unsigned int process_stack_pointer;
-extern unsigned int ksp;
+extern unsigned int ksp = &kernel_stack_pointer;
```

Comment:
Going to investigate and test removing the unnecessary 2nd level of indirection as instructed by supervisor.